.github/workflows/linux_build_main.yml

```diff
@@ -1,133 +1,150 @@
+        type: string
+      workdir:
+        required: false
+        type: string
+      prebuild_command:
+        required: false
+        type: string
+    secrets:
+      env_var_value_1:
+        required: false
+      env_var_value_2:
+        required: false
+      env_var_value_3:
+        required: false
+      env_var_value_4:
+        required: false
+      badge_service_api_key:
+        required: true
+
+jobs:
+  build:
+    name: Build main branch
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: '6.0.x'
+
+    ###############################
+    # SETUP ENVIRONMENT IF NEEDED #
+    ###############################
+    - name: Set env var 1
+      if: ${{ inputs.env_var_name_1 != '' }}
+      run: |
+        echo "::add-mask::${{ secrets.env_var_value_1 }}"
+        echo "${{ inputs.env_var_name_1 }}=${{ secrets.env_var_value_1 }}" >> $GITHUB_ENV
+        
+    - name: Set env var 2
+      if: ${{ inputs.env_var_name_2 != '' }}
+      run: |
+        echo "::add-mask::${{ secrets.env_var_value_2 }}"
+        echo "${{ inputs.env_var_name_2 }}=${{ secrets.env_var_value_2 }}" >> $GITHUB_ENV
+        
+    - name: Set env var 3
+      if: ${{ inputs.env_var_name_3 != '' }}
+      run: |
+        echo "::add-mask::${{ secrets.env_var_value_3 }}"
+        echo "${{ inputs.env_var_name_3 }}=${{ secrets.env_var_value_3 }}" >> $GITHUB_ENV
+        
+    - name: Set env var 4
+      if: ${{ inputs.env_var_name_4 != '' }}
+      run: |
+        echo "::add-mask::${{ secrets.env_var_value_4 }}"
+        echo "${{ inputs.env_var_name_4 }}=${{ secrets.env_var_value_4 }}" >> $GITHUB_ENV
+
+    - name: Set working directory if provided
+      if: ${{ inputs.workdir != '' }}
+      run: |
+        echo "workdir=${{ inputs.workdir }}" >> $GITHUB_ENV
+
+    - name: Set working directory to default if not provided
+      if: ${{ inputs.workdir == '' }}
+      run: |
+        echo "workdir=." >> $GITHUB_ENV
+
+    ###############################
+    # Install required tools      #
+    ###############################
+    - name: Install tools
+      run: |
+        sudo apt-get update
+        sudo apt-get install xmlstarlet -y
+        
+    ###############################
+    # Run prebuild command        #
+    ###############################
+    - name: Run prebuild command
+      if: ${{ inputs.prebuild_command != '' }}
+      run: ${{ inputs.prebuild_command }}
+      working-directory: "${{ env.workdir }}"
+      
+    ###############################
+    # Build, test, report results #
+    ###############################
+    - name: Build
+      run: dotnet build
+      working-directory: "${{ env.workdir }}"
+    
+    - name: Test
+      run: dotnet test --collect:"XPlat Code Coverage"
+      working-directory: "${{ env.workdir }}"
+
+    - name: Get Line Rate from output
+      id: get_line_rate
+      shell: pwsh

```

Comment:
Have you tested this? I don't think powershell works in Linux.

Reply:
This has been tested in linux build https://github.com/FrendsPlatform/Frends.PostgreSQL/actions/runs/4133197775/jobs/7142809558